### PR TITLE
PAB-4182: Select all devices

### PR DESCRIPTION
**Pull request by kailol-sag**

### CODE REVIEW

**Comment by shrutidogra-23** [comment]
More comments ..

- 6.Missing - I don't see any text like that hence my comment but you should double check this. Please let me know if you need any help fixing your tenant.

- https://softwareag-c8y-docs-build-pr-1442.surge.sh/streaming-analytics/analytics-builder/#step-3-add-the-input-block
"_The blocks for the input devices that have been registered in Cumulocity IoT are shown under Input._" -This line is incorrect? Should have been updated when we made changes to the left panel menu. It no longer shows input devices but Input blocks now. 

- I think the info box which says "_By default, the All Devices option is selected, which means that the input block is listening to all devices. However, these first steps assume that you are using your smartphone. Therefore, you must select a single device as described above._" should be above point 3 as that when the user will see it not after the user selects 3 dots.

**Comment by kailol-sag** [comment]
Missing: This is still shown - note that I removed a device on your tenant. 

image

File: content/streaming-analytics/analytics-builder-bundle/using-the-model-editor.md [hunk 1]
```diff
@@ -347,38 +346,40 @@ The following rules apply:
...
--   When you replace a group with a device:
-    -   a group is changed to a device, and
+-   When you replace a group or the **All Devices** option with a device:
+    -   a group or **All Devices**, whichever is selected, is changed to a device, and
     -   all matching trigger devices are changed to the specified device.
         {{< c8y-admon-info>}}
 If you change more than one group to a device at a time, then only the first specified device will be used to replace all trigger devices.
         {{< /c8y-admon-info>}}
--   An entry named "Trigger Device" is not available for selection in the dialog.
+-   The **Trigger Device** option is not available for selection in the dialog.
```
[the remark this answers is not in the document]
Comment (kailol-sag) [reply content/streaming-analytics/analytics-builder-bundle/using-the-model-editor.md:361]: Don't we already say this in the following bullet?

-   When you replace a device with a group or the **All Devices** option:
    -   all matching input devices are changed to groups or **All Devices**, whichever is selected, and
    -   all matching output devices are changed to trigger devices.

File: content/streaming-analytics/analytics-builder-bundle/using-the-model-editor.md [hunk 2]
```diff
@@ -347,38 +346,40 @@ The following rules apply:
...
-    -   all matching input devices are changed to groups, and
+-   You can replace a group with the **All Devices** option.
+-   You can replace the **All Devices** option with a group.
+-   When you replace a device with a group or the **All Devices** option:
+    -   all matching input devices are changed to groups or **All Devices**, whichever is selected, and
     -   all matching output devices are changed to trigger devices.
--   When you replace a group with a device:
-    -   a group is changed to a device, and
+-   When you replace a group or the **All Devices** option with a device:
+    -   a group or **All Devices**, whichever is selected, is changed to a device, and
```
[the remark this answers is not in the document]
Comment (kailol-sag) [reply content/streaming-analytics/analytics-builder-bundle/using-the-model-editor.md]: ok

File: content/streaming-analytics/analytics-builder-bundle/using-the-model-editor.md [hunk 3]
```diff
@@ -347,38 +346,40 @@ The following rules apply:
...
         {{< c8y-admon-info>}}
 If you change more than one group to a device at a time, then only the first specified device will be used to replace all trigger devices.
         {{< /c8y-admon-info>}}
--   An entry named "Trigger Device" is not available for selection in the dialog.
+-   The **Trigger Device** option is not available for selection in the dialog.
 
 After you have replaced the devices, you need to verify that the measurements that are used by the input and output blocks of the current model still refer to the appropriate measurements. The {{< product-c8y-iot >}} fragment and series are not changed by the replacement, which may or may not apply to the newly defined device.
 
-##### To replace devices, groups and assets {#to-replace-devices-groups-and-assets}
+##### To replace sources or destinations {#to-replace-sources-or-destinations}
```
[the remark this answers is not in the document]
Comment (shrutidogra-23) [reply content/streaming-analytics/analytics-builder-bundle/using-the-model-editor.md:364]: you are right it only affects template values, ignore me.

File: content/streaming-analytics/analytics-builder-bundle/using-the-model-editor.md [hunk 4]
```diff
@@ -347,38 +346,40 @@ The following rules apply:
 
 -   You can replace a device with another device.
 -   You can replace a group with another group.
--   When you replace a device with a group:
-    -   all matching input devices are changed to groups, and
+-   You can replace a group with the **All Devices** option.
```
[the remark this answers is not in the document]
Comment (kailol-sag) [reply content/streaming-analytics/analytics-builder-bundle/using-the-model-editor.md]: Done.

File: content/streaming-analytics/analytics-builder-bundle/using-the-model-editor.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (shrutidogra-23) [reply content/streaming-analytics/analytics-builder-bundle/using-the-model-editor.md:361]: Ah yes, sorry. ignore again.

**Review by MWindel — APPROVED** [review]

**Review by shrutidogra-23 — APPROVED** [review]

**Pull request merged**